src/cabal-details.js

```diff
@@ -662,6 +732,28 @@ class CabalDetails extends EventEmitter {
       })
     })
 
+    // notify when a user has archived a channel
+    this.registerListener(cabal.archives.events, 'archive', (channel, reason, key) => {
+      const user = this.users[key]
+      if (key !== this.user.key && (!user || (!user.isModerator() && !user.isAdmin()))) { return }
+      if (!this.channels[channel]) {
+        this.channels[channel] = new ChannelDetails(this.core, channel)
+      }
+      this.channels[channel].archive()
+      this._emitUpdate('channel-archive', { channel, reason, key, isLocal: key === this.user.key })
```

Comment:
declaring `isLocal = key === this.user.key` above would make this more readable. you can then use `!isLocal` in place of the same check on line 738

Reply:
good call 🖤